File: AkitaModel/Heartbeats/Heartbeat.cs
using System;

namespace Skyware.Lis.AkitaModel.Heartbeats;

/// <summary>
/// Represents a heartbeat message.
/// </summary>
public class Heartbeat
{

    /// <summary>
    /// UTC date/time of the Heartbeat message.
    /// </summary>
    public DateTime HeartbeatTime { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Local date/time of the Heartbeat message;
    /// </summary>
    public DateTime LocalHeartbeatTime => HeartbeatTime.ToLocalTime();

    /// <summary>
    /// Host where the heartbeat was generated.
    /// </summary>
    public Host Host { get; set; } = new Host();

    public string LocationId { get; set; }

    public string LocationName { get; set; }

    /// <summary>
    /// Application that generated the heartbeat.
    /// </summary>
    public ApplicationBase Application { get; set; }

}

File: AkitaMotelTS/src/Heartbeats/Heartbeat.ts
/**
 * Skyware.Lis.AkitaModel.Heartbeats
 */

import { Host } from '../Host';
import { ApplicationBase } from './ApplicationBase';

/**
 * Represents a heartbeat message.
 */
export class Heartbeat {
  /**
   * UTC date/time of the Heartbeat message.
   */
  heartbeatTime: Date = new Date();

  /**
   * Local date/time of the Heartbeat message;
   */
  get localHeartbeatTime(): Date {
    return new Date(this.heartbeatTime.toLocaleString());
  }

  /**
   * Host where the heartbeat was generated.
   */
  host: Host = new Host();

  locationId?: string;

  locationName?: string;

  /**
   * Application that generated the heartbeat.
   */
  application?: ApplicationBase;
}
